### DISCUSSION TITLE
strongswan network speed is too slow

### DISCUSSION THREAD

**Discussion by kulame** [strongswan/strongswan#1174] [Q&A]
I set up a strongswan server with v5.9.6

when i download a page on the vpn server.

```bash
root@localhost:~# wget https://assets.nflxext.com/web/ffe/wp/akira/akiraClient.js.aebe7f9798c815f1c43a.js
--2022-07-30 05:07:42--  https://assets.nflxext.com/web/ffe/wp/akira/akiraClient.js.aebe7f9798c815f1c43a.js
Resolving assets.nflxext.com (assets.nflxext.com)... 2a00:86c0:2090::1, 2a00:86c0:2091::1, 45.57.90.1, ...
Connecting to assets.nflxext.com (assets.nflxext.com)|2a00:86c0:2090::1|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/javascript]
Saving to: ‘akiraClient.js.aebe7f9798c815f1c43a.js’

akiraClient.js.aebe     [ <=>                ]   4.40M  --.-KB/s    in 0.06s   

2022-07-30 05:07:42 (69.1 MB/s) - ‘akiraClient.js.aebe7f9798c815f1c43a.js’ saved [4612511]
```

it only cost 0.06s.

but when i download the page on the vpn client.
```bash
➜  ~ wget https://assets.nflxext.com/web/ffe/wp/akira/akiraClient.js.aebe7f9798c815f1c43a.js
--2022-07-30 13:05:13--  https://assets.nflxext.com/web/ffe/wp/akira/akiraClient.js.aebe7f9798c815f1c43a.js
正在解析主机 assets.nflxext.com (assets.nflxext.com)... 45.57.91.1, 45.57.90.1, 2a00:86c0:2090::1, ...
正在连接 assets.nflxext.com (assets.nflxext.com)|45.57.91.1|:443... 已连接。
已发出 HTTP 请求，正在等待回应... 200 OK
长度： 未指定 [application/javascript]
正在保存至: ‘akiraClient.js.aebe7f9798c815f1c43a.js’

akiraClient.js.aebe     [              <=>   ]   4.40M   131KB/s    用时 1m 46s

2022-07-30 13:07:07 (42.4 KB/s) - ‘akiraClient.js.aebe7f9798c815f1c43a.js’ 已保存 [4612511]
```

it cost 1m 46s.

why?  and How can I optimize it? How can I find the bottleneck on the link？

**Comment by kulame** [comment]
can strongswan use tcpbbr to improve linux forward speed?
https://www.techrepublic.com/article/how-to-enable-tcp-bbr-to-improve-network-speed-on-linux/

**Comment by owensteam** [comment]
When I've had an issue this bad in the old days it would be because crypto being done in software though that probably isnt the case anymore but it might be something to do with fragementation? I'd try some manual icmp or tcp pings with the df bit set and see what your mtu is versus the largest packet you can actually send through might be

**Comment by Thermi** [comment]
This can be a fragmentation issue or any other. You should check perforamnce 1:1 between protocol versions and if in the VPN and outside, and check latency, ... . Many things for your to check.
